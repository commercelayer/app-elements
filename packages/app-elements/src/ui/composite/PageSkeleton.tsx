import { Container } from '#ui/atoms/Container'
import { Skeleton, SkeletonItem } from '#ui/atoms/Skeleton'
import { List } from '#ui/composite/List'
import { ListDetails } from '#ui/composite/ListDetails'
import { Report } from './Report'

export interface PageSkeletonProps {
  layout?: 'list' | 'details'
  hasHeaderDescription?: boolean
  delayMs?: number
}

function PageSkeleton({
  layout,
  hasHeaderDescription,
  delayMs
}: PageSkeletonProps): JSX.Element {
  return (
    <Container data-testid='page-skeleton'>
      <Skeleton delayMs={delayMs}>
        {/* PageHeading */}
        <div className='pt-10 pb-14'>
          <div>
            <SkeletonItem className='w-8 h-8 mb-2' />
            <SkeletonItem className='w-36 h-8' />
            {hasHeaderDescription === true && (
              <SkeletonItem
                data-testid='loading-header-description'
                className='w-36 h-5 mt-2'
              />
            )}
          </div>
        </div>

        {layout === 'list' ? (
          <List data-testid='loading-list' isLoading />
        ) : layout === 'details' ? (
          <div data-testid='loading-details'>
            <Report isLoading loadingLines={2} items={[]} />
            <ListDetails title='Details' isLoading loadingLines={4} />
          </div>
        ) : null}
      </Skeleton>
    </Container>
  )
}

PageSkeleton.displayName = 'PageSkeleton'
export { PageSkeleton }
