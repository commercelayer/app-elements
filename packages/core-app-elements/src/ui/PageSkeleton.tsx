import { Container } from './Container'
import { DetailsList } from './DetailsList'
import { ListTask } from './ListTask'
import { Report } from './Report'
import { Skeleton, SkeletonItem } from './Skeleton'

interface PageSkeletonProps {
  layout?: 'list' | 'details'
  hasHeaderDescription?: boolean
  delayMs?: number
}

export function PageSkeleton({
  layout,
  hasHeaderDescription,
  delayMs
}: PageSkeletonProps): JSX.Element {
  return (
    <Container data-test-id='page-skeleton'>
      <Skeleton delayMs={delayMs}>
        {/* PageHeading */}
        <div className='pt-10 pb-14'>
          <div>
            <SkeletonItem className='w-8 h-8 mb-2' />
            <SkeletonItem className='w-36 h-8' />
            {hasHeaderDescription === true && (
              <SkeletonItem
                data-test-id='loading-header-description'
                className='w-36 h-5 mt-2'
              />
            )}
          </div>
        </div>

        {layout === 'list' ? (
          <ListTask data-test-id='loading-list' isLoading />
        ) : layout === 'details' ? (
          <div data-test-id='loading-details'>
            <Report isLoading loadingLines={2} items={[]} />
            <DetailsList title='Details' isLoading loadingLines={4} />
          </div>
        ) : null}
      </Skeleton>
    </Container>
  )
}

export default PageSkeleton
