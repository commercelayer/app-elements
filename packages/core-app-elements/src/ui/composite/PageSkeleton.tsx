import { Container } from '#ui/atoms/Container'
import { ListDetails } from '#ui/lists/ListDetails'
import { List } from '#ui/lists/List'
import { Report } from './Report'
import { Skeleton, SkeletonItem } from '#ui/atoms/Skeleton'

interface PageSkeletonProps {
  layout?: 'list' | 'details'
  hasHeaderDescription?: boolean
  delayMs?: number
}

function PageSkeleton({
  layout,
  hasHeaderDescription,
  delayMs
}: PageSkeletonProps): JSX.Element {
  return (
    <Container data-test-id='page-skeleton'>
      <Skeleton delayMs={delayMs}>
        {/* PageHeading */}
        <div className='pt-10 pb-14'>
          <div>
            <SkeletonItem className='w-8 h-8 mb-2' />
            <SkeletonItem className='w-36 h-8' />
            {hasHeaderDescription === true && (
              <SkeletonItem
                data-test-id='loading-header-description'
                className='w-36 h-5 mt-2'
              />
            )}
          </div>
        </div>

        {layout === 'list' ? (
          <List data-test-id='loading-list' isLoading />
        ) : layout === 'details' ? (
          <div data-test-id='loading-details'>
            <Report isLoading loadingLines={2} items={[]} />
            <ListDetails title='Details' isLoading loadingLines={4} />
          </div>
        ) : null}
      </Skeleton>
    </Container>
  )
}

PageSkeleton.displayName = 'PageSkeleton'
export { PageSkeleton }
