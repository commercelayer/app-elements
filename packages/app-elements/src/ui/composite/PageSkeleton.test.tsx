import { PageSkeleton } from './PageSkeleton'
import { render } from '@testing-library/react'

describe('PageSkeleton', () => {
  test('Should be rendered', () => {
    const { getByTestId } = render(<PageSkeleton delayMs={0} />)
    const element = getByTestId('page-skeleton')
    expect(element).toBeVisible()
    expect(
      element.querySelector('loading-header-description')
    ).not.toBeInTheDocument()
  })

  test('Should have skeleton for header description ', () => {
    const { getByTestId } = render(
      <PageSkeleton hasHeaderDescription delayMs={0} />
    )
    expect(getByTestId('loading-header-description')).toBeVisible()
  })

  test('Should render a skeleton for list page', () => {
    const { getByTestId } = render(<PageSkeleton layout='list' delayMs={0} />)
    expect(getByTestId('page-skeleton')).toBeVisible()
    expect(getByTestId('loading-list')).toBeVisible()
  })

  test('Should render a skeleton for list page', () => {
    const { getByTestId } = render(
      <PageSkeleton layout='details' delayMs={0} />
    )
    expect(getByTestId('page-skeleton')).toBeVisible()
    expect(getByTestId('loading-details')).toBeVisible()
  })
})
