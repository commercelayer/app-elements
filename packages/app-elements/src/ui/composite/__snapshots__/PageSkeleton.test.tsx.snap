// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageSkeleton > Should be rendered 1`] = `
<div>
  <div
    class="select-none pointer-events-none inline"
  >
    <div
      class="select-none pointer-events-none inline"
    >
      <div
        class="container mx-auto flex flex-col px-4 md:px-0 min-h-screen"
      >
        <div
          class="mb-14"
        >
          <div
            class="select-none pointer-events-none inline"
          >
            <div
              class="w-full pt-10"
            >
              <div
                class="mb-4 flex items-center justify-between"
              >
                <button
                  class="flex items-center gap-1"
                  type="button"
                >
                  <svg
                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 256 256"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224,128a8,8,0,0,1-8,8H59.31l58.35,58.34a8,8,0,0,1-11.32,11.32l-72-72a8,8,0,0,1,0-11.32l72-72a8,8,0,0,1,11.32,11.32L59.31,120H216A8,8,0,0,1,224,128Z"
                    />
                  </svg>
                   
                  <span
                    class="font-semibold [overflow-wrap:anywhere]"
                  >
                    <span
                      class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                    >
                      Back
                    </span>
                  </span>
                </button>
              </div>
              <h1
                class="font-semibold text-title leading-title"
              >
                <span
                  class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                >
                  Loading
                </span>
              </h1>
              <div
                class="text-gray-500 leading-6 mt-2"
              />
            </div>
          </div>
          <div
            class="mb-14"
          >
            <div
              class="select-none pointer-events-none inline"
            >
              <div
                class="relative w-full"
                data-testid="SearchBar"
              >
                <div
                  class="inline-block align-middle w-fit absolute top-1/2 left-4 transform -translate-y-1/2 text-gray-400 pointer-events-none select-none text-[20px] select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M229.66,218.34l-50.07-50.06a88.11,88.11,0,1,0-11.31,11.31l50.06,50.07a8,8,0,0,0,11.32-11.32ZM40,112a72,72,0,1,1,72,72A72.08,72.08,0,0,1,40,112Z"
                    />
                  </svg>
                </div>
                <input
                  class="rounded px-11 py-2 bg-gray-100 font-medium w-full transition placeholder:text-gray-400 shadow-none !outline-0 !border-0 !ring-0 !focus:shadow-none !active:shadow-none focus:caret-primary focus:bg-white animate-pulse !bg-gray-50 placeholder:text-gray-50"
                  data-testid="SearchBar-input"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="select-none pointer-events-none inline"
          >
            <section
              aria-label="Loading"
            >
              <header
                class="border-b pb-4 flex justify-between items-center border-transparent"
              >
                <h2
                  class="text-lg font-semibold"
                >
                  <span
                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                  >
                    Loading
                  </span>
                </h2>
              </header>
              <div>
                <div>
                  <div>
                    <div
                      class=""
                    >
                      <div
                        class="flex gap-4 w-full text-gray-800 hover:text-gray-800 py-4 px-4 border-b border-gray-100"
                      >
                        <div
                          class="flex gap-4 flex-1 items-center"
                        >
                          <div
                            class="flex-shrink-0 my-0.5 self-start"
                          >
                            <div
                              class="select-none pointer-events-none inline"
                            >
                              <div
                                class="inline-block align-middle w-fit p-[10px] border rounded-full bg-gray-300 border-gray-300 text-white select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1.25rem"
                                  viewBox="0 0 256 256"
                                  width="1.25rem"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M208.49,152.49l-72,72a12,12,0,0,1-17,0l-72-72a12,12,0,0,1,17-17L116,187V40a12,12,0,0,1,24,0V187l51.51-51.52a12,12,0,0,1,17,17Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                          <div
                            class="flex justify-between w-full gap-4 items-center"
                          >
                            <div
                              class="flex-grow"
                            >
                              <div>
                                <div
                                  class="font-semibold [overflow-wrap:anywhere]"
                                >
                                  <span
                                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                                  >
                                    Loading item number 
                                  </span>
                                  <span
                                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                                  >
                                    0
                                  </span>
                                </div>
                                <div
                                  class="text-gray-500 font-medium text-sm [overflow-wrap:anywhere]"
                                >
                                  <span
                                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                                  >
                                    please wait a moment
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="text-right"
                            >
                              <div
                                class="inline-block align-middle w-fit select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  viewBox="0 0 256 256"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="flex gap-4 w-full text-gray-800 hover:text-gray-800 py-4 px-4 border-b border-gray-100"
                      >
                        <div
                          class="flex gap-4 flex-1 items-center"
                        >
                          <div
                            class="flex-shrink-0 my-0.5 self-start"
                          >
                            <div
                              class="select-none pointer-events-none inline"
                            >
                              <div
                                class="inline-block align-middle w-fit p-[10px] border rounded-full bg-gray-300 border-gray-300 text-white select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1.25rem"
                                  viewBox="0 0 256 256"
                                  width="1.25rem"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M208.49,152.49l-72,72a12,12,0,0,1-17,0l-72-72a12,12,0,0,1,17-17L116,187V40a12,12,0,0,1,24,0V187l51.51-51.52a12,12,0,0,1,17,17Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                          <div
                            class="flex justify-between w-full gap-4 items-center"
                          >
                            <div
                              class="flex-grow"
                            >
                              <div>
                                <div
                                  class="font-semibold [overflow-wrap:anywhere]"
                                >
                                  <span
                                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                                  >
                                    Loading item number 
                                  </span>
                                  <span
                                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                                  >
                                    1
                                  </span>
                                </div>
                                <div
                                  class="text-gray-500 font-medium text-sm [overflow-wrap:anywhere]"
                                >
                                  <span
                                    class="select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                                  >
                                    please wait a moment
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="text-right"
                            >
                              <div
                                class="inline-block align-middle w-fit select-none !border-gray-50 pointer-events-none animate-pulse !bg-gray-50 rounded text-transparent [&>*]:invisible object-out-of-bounds"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  viewBox="0 0 256 256"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
